CMakeLists.txt

```diff
@@ -41,61 +31,91 @@ find_package(rclpy REQUIRED)
 find_package(ros2_shared REQUIRED)
 find_package(sensor_msgs REQUIRED)
 
-include_directories(include)
-
 # Create ament index resource which references the libraries in the binary dir
 set(node_plugins "")
 
+# How to build a library:
+# https://docs.ros.org/en/humble/How-To-Guides/Ament-CMake-Documentation.html#building-a-library
+
 #=============
 # GSCam node
 #=============
 
-set(GSCAM_NODE_DEPS
-  camera_calibration_parsers
-  camera_info_manager
-  class_loader
-  rclcpp
-  rclcpp_components
-  ros2_shared
-  sensor_msgs)
-
-add_library(gscam_node SHARED)
-target_sources(gscam_node PRIVATE src/gscam_node.cpp)
+add_library(gscam_node SHARED src/gscam_node.cpp)
 add_library(gscam2::gscam_node ALIAS gscam_node)
 
 target_compile_definitions(gscam_node
   PRIVATE "COMPOSITION_BUILDING_DLL")
 
-ament_target_dependencies(gscam_node
-  ${GSCAM_NODE_DEPS})
+target_include_directories(gscam_node
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>)
+
+target_link_libraries(gscam_node
+  PRIVATE
+    class_loader::class_loader
+  PUBLIC
+    rclcpp::rclcpp
+    PkgConfig::GSTREAMER
+    PkgConfig::GST_APP)
+
+target_link_libraries(gscam_node
+  PRIVATE
+    ${rclcpp_components_TARGETS}
+    ${sensor_msgs_TARGETS})
```

Comment:
rosidl supports alias libs too since this PR:
https://github.com/ros2/rosidl/pull/605

If you need help figuring out what the target names, I can't actually tell how they work yet. Perhaps ask the maintainers, since no documentation was added on that PR (:disappointed: )

Reply:
Great! I found sensor_msgs::sensor_msgs_library by poking around the cmake files -- this does the trick. Thx.

Reply:
ah yea, in `/opt/ros/$ROS_VERSION/share/sensor_msgs/cmake/export_sensor_msgsExport.cmake:54:add_library(sensor_msgs::sensor_msgs_library INTERFACE IMPORTED)